lightning/src/ln/interactivetxs.rs

```diff
@@ -1149,19 +1219,68 @@ pub struct LocalOrRemoteInput {
 	prev_output: TxOut,
 }
 
+impl Writeable for LocalOrRemoteInput {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		write_tlv_fields!(writer, {
+			(1, self.serial_id, required),
+			(3, self.input.sequence.0, required),
+			(5, self.input.previous_output, required),
+			(7, self.input.script_sig, required),
+			(9, self.input.witness, required),
+			(11, self.prev_output, required),
+		});
+		Ok(())
+	}
+}
+
+impl Readable for LocalOrRemoteInput {
+	fn read<R: io::Read>(reader: &mut R) -> Result<Self, DecodeError> {
+		let mut serial_id: u64 = 0;
+		let mut sequence: u32 = 0;
+		let mut previous_output = OutPoint::new(Txid::from_byte_array([0; 32]), 0);
+		let mut script_sig = ScriptBuf::default();
+		let mut witness = Witness::new();
+		let mut prev_output =
+			TxOut { value: Amount::default(), script_pubkey: ScriptBuf::default() };
```

Comment:
Could we use `RequiredWrapper` instead?

Reply:
I'll `impl_consensus_ser!` for `TxIn` and then we can just reduce this to one macro as well.